lib/rucio/api/rule.py

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, Iterator, Optional
```

Comment:
`Iterator` should be imported from `collections.abc` (see https://docs.python.org/3/library/typing.html#typing.Iterator)